src/components/implementation/no_interface/cFE_booter/ostask.c

```diff
@@ -0,0 +1,815 @@
+	sl_thd_param_set(timer_thd, spperiod.v);
+	sl_thd_param_set(timer_thd, spprio.v);
+
+	sl_sched_loop();
+}
+
+void
+osal_task_entry_wrapper(void *task_entry)
+{
+	((osal_task_entry)task_entry)();
+}
+
+int
+is_valid_name(const char *name)
+{
+	int i;
+	for (i = 0; i < OS_MAX_API_NAME; i++) {
+		if (name[i] == '\0') { return TRUE; }
+	}
+	return FALSE;
+}
+
+/* TODO: Figure out how to check if thread names are taken */
+int
+is_thread_name_taken(const char *name)
+{
+	return FALSE;
+}
+
+/*
+** Task API
+*/
+
+/* NOTE: Flags here aren't implemented, but I can't find an implementation that does */
+int32
+OS_TaskCreate(uint32 *task_id, const char *task_name, osal_task_entry function_pointer, uint32 *stack_pointer,
+              uint32 stack_size, uint32 priority, uint32 flags)
+{
+	/* TODO: Verify that we don't need to take the cs here */
+
+	if (task_id == NULL || task_name == NULL || function_pointer == NULL) { return OS_INVALID_POINTER; }
+
+	if (!is_valid_name(task_name)) { return OS_ERR_NAME_TOO_LONG; }
+
+	if (is_thread_name_taken(task_name)) { return OS_ERR_NAME_TAKEN; }
+
+	if (priority > 255 || priority < 1) { return OS_ERR_INVALID_PRIORITY; }
+
+	struct sl_thd *thd = sl_thd_alloc(osal_task_entry_wrapper, function_pointer);
+	assert(thd);
+	union sched_param_union sp = {.c = {.type = SCHEDP_PRIO, .value = priority}};
+	sl_thd_param_set(thd, sp.v);
+
+	struct cfe_task_info *task_info = &cfe_tasks[sl_thd_thdid(thd)];
+	strcpy(task_info->osal_task_prop.name, task_name);
+	task_info->osal_task_prop.creator    = OS_TaskGetId();
+	task_info->osal_task_prop.stack_size = stack_size;
+	task_info->osal_task_prop.priority   = priority;
+	task_info->osal_task_prop.OStask_id  = (uint32)sl_thd_thdid(thd);
+	task_info->delete_handler            = NULL;
+
+	*task_id = (uint32)sl_thd_thdid(thd);
+
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskDelete(uint32 task_id)
+{
+	struct sl_thd *thd = sl_thd_lkup(task_id);
+	if (!thd) { return OS_ERR_INVALID_ID; }
+
+	struct cfe_task_info *task_info = &cfe_tasks[task_id];
+
+	osal_task_entry delete_handler = task_info->delete_handler;
+	if (delete_handler) { delete_handler(); }
+
+	sl_thd_free(thd);
+
+	return OS_SUCCESS;
+}
+
+uint32
+OS_TaskGetId(void)
+{
+	thdid_t real_id           = sl_thdid();
+	thdid_t possible_override = id_overrides[real_id];
+	if (possible_override) return possible_override;
+	return real_id;
+}
+
+void
+OS_TaskExit(void)
+{
+	sl_thd_free(sl_thd_curr());
+	PANIC("Should be unreachable!");
+}
+
+int32
+OS_TaskInstallDeleteHandler(osal_task_entry function_pointer)
+{
+	if (OS_TaskGetId() == main_delegate_thread_id) { return OS_ERR_INVALID_ID; }
+
+	struct cfe_task_info *task_info = &cfe_tasks[sl_thd_thdid(sl_thd_curr())];
+	task_info->delete_handler = function_pointer;
+
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskDelay(uint32 millisecond)
+{
+	cycles_t wakeup = sl_now() + sl_usec2cyc(millisecond * 1000);
+	sl_thd_block_timeout(0, wakeup);
```

Comment:
empty lines.

Reply:
wow, relative task delay APIs are broken.  Way to go cfe.  (though Posix made the same mistake).